### PR TITLE
usbip driver install from command line

### PR DESCRIPTION
**Pull request by koudis**
First, working version of USBIP driver install from command line - `usbip install`

History is not referencable. I will squash all commits into one after PR approve.

### CODE REVIEW

File: userspace/src/usbip/usbip.c [hunk 1]
```diff
@@ -50,47 +50,53 @@ struct command {
 
 static const struct command cmds[] = {
 	{
-		.name  = "help",
```
Comment (koudis) [inline userspace/src/usbip/usbip.c]: I do not known why are these changes here. I created my branch from updatream master.

Comment (koudis) [reply userspace/src/usbip/usbip.c]: I noticed now. It's Visual Studio feature. I will repair.

File: userspace/src/usbip/usbip_install.c [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+static struct usbip_install_devinfo_struct device_list[] = {
+	{
+		.inf_filename = "usbip_vhci.inf",
+		.hwid_inf_section = "Standard.NTamd64",
+		.hwid_inf_key = "USB/IP VHCI",
+		.devdesc_inf_section = "Strings",
+		.devdesc_inf_key = "DeviceDesc",
+		.dev_instance_path = "ROOT\\USBIP\\0000"
+	}
```
Comment (koudis) [inline userspace/src/usbip/usbip_install.c]: I am not sure if values can stay here. Maybe move to some header file?

File: userspace/src/usbip/usbip.vcxproj [hunk 3]
```diff
@@ -109,7 +109,7 @@
     </ClCompile>
     <Link>
       <GenerateDebugInformation>true</GenerateDebugInformation>
-      <AdditionalDependencies>setupapi.lib;advapi32.lib;ws2_32.lib;wintrust.lib;crypt32.lib;newdev.lib</AdditionalDependencies>
+      <AdditionalDependencies>Shlwapi.lib;setupapi.lib;advapi32.lib;ws2_32.lib;wintrust.lib;crypt32.lib;newdev.lib</AdditionalDependencies>
```
Comment (koudis) [inline userspace/src/usbip/usbip.vcxproj]: I use this for path parsing and manipulation.

**Comment by koudis** [comment]
Not final version. Do not make code review please. Just for now.
I will repair some coding standard constraint etc.

**Comment by xinheng0816** [comment]
Hello all
Sorry for interupt!   
Someone can guide me how to install compile and install usbip driver in RedHat system ?
BTW, I downlod all source code with verison = Linux version 3.10.0-514.el7.x86_64.   When I tied to build usbip module,  it seems always nothing happen and the expected kos (usbip-core.ko & usbip-host.ko) can't be generated.   
Thank you all!

**Comment by xinheng0816** [comment]
Let me describe my quesiton more clearly.
Precondition:
OS: RedHat 7.3
Linux kenel: Linux version 3.10.0-514.el7.x86_64

Background:
As required, i need to setup usbip env in my Linux station, so I'm considerring to build usbip source code locally.

Problem:
Until now, I downloaded the specified linux kernel source code (3.10.0-514.el7). Following https://wiki.centos.org/HowTos/BuildingKernelModules, when I execute below command under root directory of linux kernel source, I can't build out the expected Ko files.
$ make
$ make M=drivers/staging
Expected result:
usbip-host.ko and usbip-core.ko can be generated
Actual result:
From build log, I saw drivers/staging/usbip wasn't built at all.

Thanks for you time and expect your instruction!!

**Comment by koudis** [comment]
@xinheng0816 Sorry but i am not able to help you. There are many tutorial how to build specific dirvers for linux kernel. Maybe RedHat official doc?

I close the PR because I merge my chages to #90
